### PR TITLE
feat(bux-399): Confirm contact

### PR DESCRIPTION
**Pull request by Nazarii-4chain**
Added confirm Contact PIKE
# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [x] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

**Review by chris-4chain — CHANGES_REQUESTED** [review]

File: actions/contacts/routes.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ func NewHandler(appConfig *config.AppConfig, services *config.AppServices) route
 	apiEndpoints := routes.APIEndpointsFunc(func(router *gin.RouterGroup) {
 		contactGroup := router.Group("/contact")
 		contactGroup.POST("", action.create)
+		contactGroup.PATCH("", action.update)
```
[the remark this answers is not in the document]
Comment (Nazarii-4chain) [reply actions/contacts/routes.go:22]: BUX-399 supposed to only change contact status to confirmed.

File: engine/action_contact.go [hunk 2]
```diff
@@ -54,6 +54,40 @@ func (c *Client) NewContact(ctx context.Context, fullName, paymail, senderPubKey
...
+		contact.PubKey = pubKey
+	}
+
+	if status != "" {
+		contact.Status = status
+	}
+
+	if paymailAddr != "" {
+		contact.Paymail = paymailAddr
+	}
```
[the remark this answers is not in the document]
Comment (Nazarii-4chain) [reply engine/action_contact.go:82]: Not sure about that. I left an option to change contact's paymail.

File: actions/contacts/update.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+// @Tags		Contacts
+// @Produce		json
+// @Param		metadata body string true "Contacts Metadata"
+// @Success		200
+// @Router		/v1/contact [patch]
+// @Security	x-auth-xpub
+func (a *Action) update(c *gin.Context) {
+	reqXPubID := c.GetString(auth.ParamXPubHashKey)
+
+	var requestBody UpdateContact
```
[the remark this answers is not in the document]
Comment (Nazarii-4chain) [reply actions/contacts/update.go:25]: Thanks. Added sanitization.

File: actions/contacts/models.go [hunk 4]
```diff
@@ -1,8 +1,21 @@
...
 	Metadata engine.Metadata `json:"metadata"`
 }
+
+// UpdateContact is the model for updating a contact
+type UpdateContact struct {
+	XPubID   string               `json:"xpub_id"`
+	FullName string               `json:"full_name"`
+	Paymail  string               `json:"paymail"`
+	PubKey   string               `json:"pubKey"`
+	Status   models.ContactStatus `json:"status"`
```
[the remark this answers is not in the document]
Comment (Nazarii-4chain) [reply actions/contacts/models.go:19]: Could you add more details what do you mean ?

File: actions/contacts/models.go [hunk 5]
```diff
@@ -1,8 +1,21 @@
...
 // CreateContact is the model for creating a contact
 type CreateContact struct {
 	Metadata engine.Metadata `json:"metadata"`
 }
+
+// UpdateContact is the model for updating a contact
+type UpdateContact struct {
+	XPubID   string               `json:"xpub_id"`
+	FullName string               `json:"full_name"`
+	Paymail  string               `json:"paymail"`
```
[the remark this answers is not in the document]
Comment (chris-4chain) [reply actions/contacts/models.go:17]: I see this helper func as a DTO (Data Transfer Object) which - I think - should have all model which is sent as JSON. 
My suggestion is to skip getting fields explicitly (using e.g. GetString func) and use only parsed Json model.

image

File: actions/contacts/models.go [hunk 6]
```diff
@@ -1,8 +1,21 @@
...
+	PubKey   string               `json:"pubKey"`
```
[the remark this answers is not in the document]
Comment (chris-4chain) [reply actions/contacts/models.go:18]: We definitely should add a comment to both fields to clarify what they are.

**Review by chris-4chain — APPROVED** [review]

**Pull request merged**